src/components/common/Directors.vue

```diff
@@ -845,32 +854,26 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
     this.draftDate = date
   }
 
-  private getOriginalDirectors () {
-    this.getDirectors(true)
-  }
-
   /**
    * Function called internally and externally to fetch the list of directors.
-   * FUTURE: change this to a prop?
    */
-  public getDirectors (getOrigOnly: Boolean = false): void {
+  public async getDirectors (getOrigOnly: boolean = false): Promise<void> {
```

Comment:
This was pretty important to the bugs here. Since this was not async (and not awaited by its callers), some things would sometimes happen in different orders (ie, timing bug).